endpoint.go

```diff
@@ -456,15 +456,23 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 		if extEp != nil {
 			log.Debugf("Revoking external connectivity on endpoint %s (%s)", extEp.Name(), extEp.ID())
 			if err := d.RevokeExternalConnectivity(extEp.network.ID(), extEp.ID()); err != nil {
-				log.Warnf("driver failed revoking external connectivity on endpoint %s (%s): %v",
+				return types.InternalErrorf(
+					"driver failed revoking external connectivity on endpoint %s (%s): %v",
 					extEp.Name(), extEp.ID(), err)
 			}
+			defer func() {
+				if err := d.ProgramExternalConnectivity(extEp.network.ID(), extEp.ID(), sb.Labels()); err != nil {
```

Comment:
What is the need to unconditionally call `ProgramExternalConnectivity` in the defer ?
That seems incorrect both to the fact that we are doing it unconditionally and also to the fact that the error check cannot be performed with the way `err` variable is managed later and `return sb.clearDefaultGW()` is called without updating the `err` variable.